### PR TITLE
Disconnect automatically configured webhook after disconnecting or changing account

### PR DESCRIPTION
**Pull request by james-allan**
Part of https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3195

## Changes proposed in this Pull Request:

When a merchant changes or deletes their API keys, the automatically configured webhook should be decommissioned and deleted as it is no longer relevant. 

This has the benefit of deleting webhooks on the Stripe account that are no longer in use, and avoids situations where a merchant could configure webhooks, change accounts and the **"Webhooks Enabled"** flag would confuse them into thinking the webhooks are set up, when in fact the webhook secret is linked to an old/different account. 

## Testing instructions

**Preconfigured flow**

1. On `develop` go to **WooCommerce > Settings > Payments > Stripe > Settings**.
2. In the _accounts keys modal_ add your Stripe API keys including a webhook secret. 
3. Checkout this branch. 
4. Run `npm run build` as this branch includes built file changes.
5. Change your API keys to a different account. 
6. Note that the previously added webhook secret is still set.

> [!IMPORTANT]
> **Resolved** - The webhook secret is now deleted if you change keys. ~~**Question**: Should we remove the webhook secret stored in settings in this case too? It kinda makes sense to me that we'd also unset the webhook secret even if we can't delete it at Stripe when you change Stripe Accounts. Curious for your thoughts.~~

**Automatically configured flow**

1. Click the "Reconfigure webhooks" button on the API credentials modal. 
2. Now change your API keys to a different account or remove them entirely.
3. Save the keys. 
6. When the page finishes loading you'll notice the previously configured webhook has been deleted.
   - On `develop` or `fix/3195-autogenerate-webhook-secret` the webhook wouldn't have been deleted. 

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [ ] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

### CODE REVIEW

File: client/data/account-keys/actions.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ export function* saveAccountKeys( accountKeys ) {
 			data: accountKeys,
 		} );
 
-		if ( ! accountData?.id ) {
+		if ( ! isDisconnecting && ! accountData?.id ) {
```
Comment (james-allan) [inline client/data/account-keys/actions.js:58]: I noticed this error when testing this PR so snuck a fix into this PR.

To replicate the issue this change is fixing:

1. On `develop` go to **WooCommerce > settings > Payments > Stripe > Settings**
2. Open the Account keys modal. 
3. Delete your keys. 
4. Save.
5. On develop you will get this error:
<p align="center">
Screenshot 2024-06-25 at 5 07 48 PM
</p>

It turns out this error was caused by this `throw 'Account not Found';`. When disconnecting the account, no account data being returned is the expected outcome and so no error should be thrown in that case. 

On this branch, if you follow those steps, you will see this successful message instead:
<p align="center">
Screenshot 2024-06-25 at 5 11 03 PM
</p>

**Review by a-danae — APPROVED** [review]
Happy to see this implemented! Thanks for these changes, @james-allan ! 

✅ Changing the account keys after automatically having configured deletes the webhook in the Stripe account and its data in the store settings
✅ There's no longer an error message when disconnecting the account
✅ The code makes sense to me

> Question: Should we remove the webhook secret stored in settings in this case too? It kinda makes sense to me that we'd also unset the webhook secret even if we can't delete it at Stripe when you change Stripe Accounts. Curious for your thoughts.

I agree. Removing the webhook secret stored in the settings sounds good to me!

Thanks again!

File: client/data/account-keys/actions.js (hunk 1 above)
Comment (a-danae) [reply client/data/account-keys/actions.js:58]: Niiiice. Good thing to sneak in. Working great, thankya!

**Comment by james-allan** [comment]
> I agree. Removing the webhook secret stored in the settings sounds good to me!

OK sounds good. I've move the logic around so if you previously entered a webhook secret key manually, changing the API keys deletes that webhook secret. As I think we both agree, that makes sense given that webhook secret is tied to a different account and isn't relevant anymore.

**Comment by james-allan** [comment]
> OK sounds good. I've move the logic around so if you previously entered a webhook secret key manually, changing the API keys deletes that webhook secret

I've had to revert this change because it broke too many flows. I don't think it's critical though given I don't suspect people change their keys so frequently.

**Pull request merged**